hack/bundle-gen.py

```diff
@@ -0,0 +1,410 @@
+                        action='store_true')
+    parser.add_argument('--github-user',
+                        default=os.environ['USER'],
+                        help="User's github username, if different than $USER")
+    parser.add_argument('--hold',
+                        default=False,
+                        help='Adds a /hold comment in commit body to prevent the PR from merging (use "/hold cancel" to remove)',
+                        action='store_true')
+    args = parser.parse_args()
+
+    if args.verbose:
+        global SUBPROCESS_REDIRECT
+        SUBPROCESS_REDIRECT = None
+
+    return args
+
+def validate_params(args):
+    if not os.path.isdir(args.hive_repo_dir):
+        print('Default repo-dir path ({}) does not exist, provide --hive-repo-dir'.format(args.hive_repo_dir))
+        sys.exit(1)
+    if not os.path.isfile(args.registry_auth_file):
+        print('Default registry-auth-file ({}) does not exist, provide --registry-auth-file'.format(args.registry_auth_file))
+        sys.exit(1)
+
+# build_and_push_image uses buildah to build the HIVE image (tagged with image_tag) and then pushes the image to quay
+def build_and_push_image(repo_dir, registry_auth_file, image_tag, commit_hash):
+    prev_dir = os.getcwd()
+    os.chdir(repo_dir)
+
+    try:
+        # sync up git repo with remote
+        cmd = "git fetch upstream".split()
+        subprocess.run(cmd)
+        cmd = "git fetch upstream --tags".split()
+        subprocess.run(cmd)
+
+        # checkout tag/commit
+        cmd = 'git checkout {}'.format(commit_hash).split()
+        subprocess.run(cmd)
+
+        # build/push the thing
+        print("Building container")
+        container_name = 'quay.io/openshift-hive/hive:{}'.format(image_tag)
+        cmd = 'buildah bud --tag {} -f ./Dockerfile'.format(container_name).split()
+        subprocess.run(cmd)
+
+        print("Pushing container")
+        cmd = 'buildah push '
+        if registry_auth_file != None:
+            cmd = cmd + ' --authfile={} '.format(registry_auth_file)
+        cmd = cmd + ' {}'.format(container_name)
+        subprocess.run(cmd.split())
+    finally:
+        os.chdir(prev_dir)
+
+# gen_hive_version generates and returns the hive version eg. "v1.2.3187-18827f6"
+def gen_hive_version(repo_dir, commit_hash):
+    repo = git.Repo(repo_dir)
+
+    # sha is the first 7 characters of commit_hash
+    sha = str(repo.rev_parse(commit_hash))[0:7]
+
+    # num commits is the number of git commits counted from the first commit to the provided commit_hash
+    # this is the equivalent of running:
+    # git rev-list `git rev-list --parents HEAD | egrep "^[a-f0-9]{40}$"`..HEAD --count
+    parent = repo.git.rev_list('--max-parents=0', commit_hash)
+    num_commits = repo.git.rev_list('--count', '{parent}..{commit_hash}'.format(parent=parent, commit_hash=commit_hash))
+
+    return '{prefix}.{commits}-{sha}'.format(prefix=HIVE_VERSION_PREFIX, commits=num_commits, sha=sha)
```

Comment:
For older branches, it won't be`HIVE_VERSION_PREFIX`. That's where we'll want to do something like `branch_name.split('-', 1)[0]`